Spatial.gd

```diff
@@ -397,69 +401,20 @@ func gen_map_markers():
-		var path_points := PoolVector3Array()
-		var trail_data = path.get_trail_data()
-		if trail_data.get_points_x().size() != trail_data.get_points_y().size() or trail_data.get_points_x().size() != trail_data.get_points_z().size():
-			print("Warning: Trail ", path.get_category.get_name(), " does not have equal number of X, Y, and Z coordinates.")
-		for index in range(0, trail_data.get_points_z().size()):
-			path_points.append(Vector3(trail_data.get_points_x()[index], trail_data.get_points_y()[index], trail_data.get_points_z()[index]))
-		var texture_path = path.get_texture_path()
-		if texture_path == null:
-			print("Warning: No texture found in " , path.get_category().name())
-			continue
-		var full_texture_path = self.marker_file_dir + texture_path.get_path()
-		#TODO This tree look up is unnescary if we make path a child of Category in protobuf
-		var split_name = path.get_category().get_name().split(".")
-		var category_item = search_category_tree(split_name, self.marker_packs.get_root())
-		gen_new_path(path_points, full_texture_path, path, category_item)
-	for icon in self.markerdata.get_icon():
-		var position = icon.get_position()
-		if position == null:
-			print("Warning: No position found for icon ", icon.get_category().name())
-			continue
-		var position_vector = Vector3(position.get_x(), position.get_y(), position.get_z())
-		var texture_path = icon.get_texture_path()
-		if texture_path == null:
-			print("Warning: No texture found in " , icon.get_category().name())
-			continue
-		var full_texture_path = self.marker_file_dir + texture_path.get_path()
-		#TODO This tree look up is unnescary if we make path a child of Category in protobuf
-		var split_name = icon.get_category().get_name().split(".")
-		var category_item = search_category_tree(split_name, self.marker_packs.get_root())
-		gen_new_icon(position_vector, full_texture_path, icon, category_item)
-
-
-func search_category_tree(split_name: PoolStringArray, category_item: TreeItem, index: int = 0):
-	if index == split_name.size():
-		return category_item
-	var child_item = category_item.get_children()
-	while child_item != null:
-		var joined_name = ""
-		for i in range(index):
-			joined_name += split_name[i]
-		if child_item.get_metadata(0) == joined_name:
-			return search_category_tree(split_name, child_item, index + 1)
-		child_item = child_item.get_next()
-	print("No category found for ", split_name)
-	return null
 
-func build_category_tree():
-	var root = self.marker_packs.create_item()
+func init_category_tree():
+	self.root = self.marker_packs.create_item()
 	root.set_text(0, "Markers available on current map")
 	root.set_selectable(0, false)
 	root.set_text(1, "Visible")
-	
-	for category in self.markerdata.get_category():
-		self.marker_packs_array.append(category.get_name())
-		add_category(root, category, category.get_name(), false)
 
 
-func add_category(item: TreeItem, category, full_category_name: String, collapsed: bool):
+func parse_Waypoint():
```

Comment:
This might want to be something more descriptive to indicate that it is creating nodes from the waypoint category object.